### PR TITLE
Change to make Read compatible with Shapeless 2.1 and higher

### PR DESCRIPTION
**Pull request by Kevin-Lee**

### CODE REVIEW

**Comment by Kevin-Lee** [comment]
Build for Scala 2.10 will pass once #8 is merged and rebased.

File: src/main/scala/pirate/Read.scala [hunk 1]
```diff
@@ -180,7 +180,14 @@ object Read extends shapeless.ProductTypeClassCompanion[Read] {
...
 
-  implicit def ReadTypeClass: shapeless.ProductTypeClass[Read] = new shapeless.ProductTypeClass[Read] {
+  /*
+   * WARNING!!! Don't change the name `typeClass`.
+   * It has to be `typeClass` to be compatible with Shapeless 2.1 and higher.
+   *
+   * Reference:
+   * https://github.com/milessabin/shapeless/blob/shapeless-2.1.0/core/src/main/scala/shapeless/typeclass.scala#L48
+   */
+  implicit val typeClass: shapeless.ProductTypeClass[Read] = new shapeless.ProductTypeClass[Read] {
```
Comment (charleso) [inline src/main/scala/pirate/Read.scala]: Not sure you need such a large warning. Just add `override` and remove the comment?

Comment (Kevin-Lee) [reply src/main/scala/pirate/Read.scala]: It's not `override` unless Shapeless is upgraded to `2.1.0` or higher. Should I upgrade Shapeless from `2.0.0` to `2.1.0`?

Comment (Kevin-Lee) [reply src/main/scala/pirate/Read.scala]: As discussed, I'll upgrade it to `2.1.0`.

File: src/test/scala/pirate/ReadSpec.scala [hunk 2]
```diff
@@ -104,9 +104,6 @@ object ReadSpec extends Properties {
   Read.of[java.net.URI]
   Read.of[java.net.URL]
   Read.of[Option[String]]
-
-  import Read.auto._
```
Comment (charleso) [inline src/test/scala/pirate/ReadSpec.scala]: Looks like this broke 2.10? Why the change?

Comment (Kevin-Lee) [reply src/test/scala/pirate/ReadSpec.scala]: `Read.auto` does not exist in Shapeless `2.1.0`. Sorry I should have run `sbt +test` locally. I'll fix it.

Comment (Kevin-Lee) [reply src/test/scala/pirate/ReadSpec.scala]: Yeah if I just do this for `Read.of[(Int, Int)]`, there's no error for `Read.of[(Int, Int)]`.

```scala
Read.of[(Int, Int)](tryRead(_ => (1, 1), "(Int, Int)"))
```
